src/plugins/util/components.js

```diff
@@ -94,4 +98,116 @@ const switchButton = (() => {
     return {createSwitch};
 })();
 
-module.exports = {switchButton};
+/**
+ * Creates a custom slider component to set a value between a range.
+ * @function
+ * @param {Object} options - Options for configuring the slider.
+ * @param {Function} [options.onChange] - Optional callback function to be executed when the slider value changes.
+ * @returns {HTMLElement} - The slider component as an HTML element. setValue function is exposed to change the value of the slider.
+ */
+
+const slider = (() => {
+    const createSlider = ({onChange = null, onCursorMove = null, min = 0, max = 100, value = 50}) => {
+        // Validate parameters
+        if (min >= max) {
+            throw new Error('`min` must be less than `max`.');
+        }
+        if (value < min || value > max) {
+            throw new Error('`value` must be within the range defined by `min` and `max`.');
+        }
+
+        // Container for the slider
```

Comment:
Some thing that might be useful, in general, could be to have a comment of what the generated html looks like.

But this is an idea in general about the project, not specific to the slider changes here :) 

Something to discuss/think about for later.